src/biz/collab/ops.rs

```diff
@@ -498,6 +476,142 @@ pub async fn list_database_row_ids(
   Ok(db_rows)
 }
 
+pub async fn insert_database_row(
+  collab_storage: &CollabAccessControlStorage,
+  pg_pool: &PgPool,
+  workspace_uuid_str: &str,
+  database_uuid_str: &str,
+  uid: i64,
+  cell_value_by_id: HashMap<String, serde_json::Value>,
+) -> Result<String, AppError> {
+  let mut db_txn = pg_pool.begin().await?;
```

Comment:
can move to line 547?

Reply:
I'll move it down to about 543, as this op also requires inserting row collab, which should use the same transaction

Reply:
Some additional modification: I have move all the collab non db ops to the top, which further shorten the txn duration.